src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -73099,6 +73099,307 @@ static void Main()
+            Assert.Equal("C<T?>", c3.ToTestDisplayString());
+            Assert.False(c3.IsDefinition);
+
+            AssertHashCodesMatch(cDefinition, c3);
+
+            var i3 = c3.GetTypeMember("I");
+            Assert.Equal("C<T?>.I", i3.ToTestDisplayString());
+            Assert.False(i3.IsDefinition);
+
+            AssertHashCodesMatch(iDefinition, i3);
+        }
+
+        [Fact]
+        [WorkItem(30673, "https://github.com/dotnet/roslyn/issues/30673")]
+        public void TypeSymbolGetHashCode_ContainingType_Generic()
+        {
+            var text = @"
+class C<T> where T : class
+{
+    interface I<U> where U : class { }
+}
+";
+
+            var comp = CreateNullableCompilation(text);
+            var iDefinition = comp.GetMember<NamedTypeSymbol>("C.I");
+            Assert.Equal("C<T>.I<U>", iDefinition.ToTestDisplayString());
+            Assert.True(iDefinition.IsDefinition);
+
+            var cDefinition = iDefinition.ContainingType;
+            Assert.Equal("C<T>", cDefinition.ToTestDisplayString());
+            Assert.True(cDefinition.IsDefinition);
+
+            var c2 = cDefinition.Construct(ImmutableArray.Create(TypeWithAnnotations.Create(cDefinition.TypeParameters.Single(), NullableAnnotation.NotAnnotated)));
+            var i2 = c2.GetTypeMember("I");
+            Assert.Equal("C<T>.I<U>", i2.ToTestDisplayString());
+            Assert.True((object)i2.OriginalDefinition == iDefinition);
+            AssertHashCodesMatch(i2, iDefinition);
+
+            var i2constructed = i2.Construct(iDefinition.TypeParameters.Single());
+            Assert.Equal("C<T>.I<U>", i2constructed.ToTestDisplayString());
+            AssertHashCodesMatch(iDefinition, i2constructed);
+
+            var i2b = i2.Construct(ImmutableArray.Create(TypeWithAnnotations.Create(i2.TypeParameters.Single(), NullableAnnotation.Annotated)));
+            Assert.Equal("C<T>.I<U?>", i2b.ToTestDisplayString());
+            AssertHashCodesMatch(iDefinition, i2b);
+
+            var c3 = cDefinition.Construct(ImmutableArray.Create(TypeWithAnnotations.Create(cDefinition.TypeParameters.Single(), NullableAnnotation.Annotated)));
+            var i3 = c3.GetTypeMember("I");
+            Assert.Equal("C<T?>.I<U>", i3.ToTestDisplayString());
+            AssertHashCodesMatch(iDefinition, i3);
+
+            var i3b = i3.Construct(ImmutableArray.Create(TypeWithAnnotations.Create(i3.TypeParameters.Single(), NullableAnnotation.Annotated)));
+            Assert.Equal("C<T?>.I<U?>", i3b.ToTestDisplayString());
+            AssertHashCodesMatch(iDefinition, i3b);
+        }
+
+        private static void AssertHashCodesMatch(TypeSymbol c, TypeSymbol c2)
+        {
+            Assert.False(c.Equals(c2));
+            Assert.True(c.Equals(c2, TypeCompareKind.CLRSignatureCompareOptions));
+
+            Assert.True(c2.GetHashCode() == c.GetHashCode(), "hash codes differed even though types matched (ignoring nullability)");
```

Comment:
>Assert.True [](start = 12, length = 11)

Can we use ```Assert.Equal``` instead?

Reply:
I chose to use `Assert.True` so that I could put a message as a comment.

---
In reply to: [283574566](https://github.com/dotnet/roslyn/pull/35613#discussion_r283574566) [](ancestors = 283574566)